app/models/manageiq/providers/cisco_intersight/inventory/parser/physical_infra_manager.rb

```diff
@@ -437,6 +445,22 @@ def build_decomissioned_physical_infrastructure(server)
       )
     end
 
+    def build_physical_server_profiles(physical_server_profile)
+      # Builds out collection physical_server_profiles
+      # Object types:
+      #   - physical_server_profile - ServerProfile, object obtained by intersight client
+      # Returns:
+      #   ManageIQ's object ManageIQ::Providers::CiscoIntersight::PhysicalInfraManager::PhysicalServerProfile
+      assigned_server = physical_server_profile.assigned_server.nil? ? nil : persister.physical_servers.lazy_find(physical_server_profile.assigned_server.moid)
+      associated_server = physical_server_profile.associated_server.nil? ? nil : persister.physical_servers.lazy_find(physical_server_profile.associated_server.moid)
```

Comment:
`#lazy_find` actually returns `nil` if the manager_ref passed in is `nil`: https://github.com/ManageIQ/inventory_refresh/blob/master/lib/inventory_refresh/inventory_collection/index/proxy.rb#L113

```suggestion
      assigned_server = persister.physical_servers.lazy_find(physical_server_profile.assigned_server&.moid)
      associated_server = persister.physical_servers.lazy_find(physical_server_profile.associated_server&.moid)
```